src/Symfony/Component/Dsn/Tests/Factory/MemcachedConnectionFactoryTest.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Tests\Adapter;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Dsn\Factory\MemcachedConnectionFactory;
+
+class MemcachedConnectionFactoryTest extends TestCase
+{
+    public function testOptions()
+    {
+        $client = MemcachedConnectionFactory::createConnection(array(), array(
+            'libketama_compatible' => false,
+            'distribution' => 'modula',
+            'compression' => true,
+            'serializer' => 'php',
+            'hash' => 'md5',
+        ));
+
+        $this->assertSame(\Memcached::SERIALIZER_PHP, $client->getOption(\Memcached::OPT_SERIALIZER));
+        $this->assertSame(\Memcached::HASH_MD5, $client->getOption(\Memcached::OPT_HASH));
+        $this->assertTrue($client->getOption(\Memcached::OPT_COMPRESSION));
+        $this->assertSame(0, $client->getOption(\Memcached::OPT_LIBKETAMA_COMPATIBLE));
+        $this->assertSame(\Memcached::DISTRIBUTION_MODULA, $client->getOption(\Memcached::OPT_DISTRIBUTION));
+    }
+
+    /**
+     * @dataProvider provideBadOptions
+     * @expectedException \ErrorException
+     * @expectedExceptionMessage constant(): Couldn't find constant Memcached::
```

Comment:
using annotations is considered as bad practice for expectations.
please use `->expectException` (or `->setExpectedException` if lower phpunit is needed)

Reply:
The issue is we would have to use both of them (as lower PHPUnit is needed for older PHP versions, but PHPUnit 6 removed `expectException` entirely).
The annotation works in all versions.

And we use them in most places in the Symfony testsuite btw.

Reply:
PHPUnit 6 removed `setExpectedException`, not `expectException`.
Is there a need to support PHPUnit 4, 5, and 6 ?

Reply:
yes, due to PHP versions being supported (PHPUnit 5 does not support PHP 5.5, which is our min version in Symfony 3.x, and so must be covered in CI)